yktrace/src/sir.rs

```diff
@@ -193,32 +237,34 @@ pub fn sir_trace_str<'a>(trace: &'a dyn SirTrace, trimmed: bool, show_blocks: bo
 
 impl Display for dyn SirTrace {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", sir_trace_str(self, true, true))
+        write!(f, "{}", sir_trace_str(&*SIR, self, false, true))
     }
 }
 
 /// An iterator over a trimmed SIR trace.
 pub struct SirTraceIterator<'a> {
+    sir: &'a Sir,
     trace: &'a dyn SirTrace,
     next_idx: usize
 }
 
 impl<'a> SirTraceIterator<'a> {
-    fn new(trace: &'a dyn SirTrace) -> Self {
+    pub fn new(sir: &'a Sir, trace: &'a dyn SirTrace) -> Self {
         // We are going to present a "trimmed trace", so we do a backwards scan looking for the end
         // of the code that starts the tracer.
         let mut begin_idx = None;
         for blk_idx in (0..trace.raw_len()).rev() {
             let sym = &trace.raw_loc(blk_idx).symbol_name;
-            if SIR.markers.trace_heads.contains(sym) {
+            if sir.markers.trace_heads.contains(sym) {
                 begin_idx = Some(blk_idx + 1);
                 break;
             }
         }
 
         SirTraceIterator {
+            sir,
             trace,
-            next_idx: begin_idx.expect("Couldn't find the end of the code that starts the tracer")
+            next_idx: begin_idx.unwrap_or(0),//.expect("Couldn't find the end of the code that starts the tracer")
```

Comment:
I'd prefer the crash.

Reply:
That change shouldn't have been commited.